src/lib/Search/Search.spec.ts

```diff
@@ -0,0 +1,312 @@
+    });
+    const container = getByTestId(searchContainerTestId);
+    const style = window.getComputedStyle(container);
+    expect(style.getPropertyValue("--outline-color")).toBe("magenta");
+  });
+
+  test("inputId prop", async () => {
+    const { getByRole } = render(Search, {
+      target,
+      props: {
+        inputId: "custom-id",
+        suggestions
+      }
+    });
+    const input = getByRole("combobox");
+    await fireEvent.focus(input);
+    const label = document.getElementById("cac-label-custom-id");
+    const listbox = getByRole("listbox");
+    expect(input.getAttribute("id")).toBe("custom-id");
+    expect(label).toBeTruthy();
+    expect(listbox.getAttribute("id")).toBe("cac-listbox-custom-id");
+  });
+
+  test("clearSearch fn prop", async () => {
+    const { getByRole, component } = render(Search, {
+      target,
+      props: {
+        suggestions
+      }
+    });
+    const input = getByRole("combobox") as HTMLInputElement;
+    await fireEvent.focus(input);
+    await component.$set({ searchValue: "foo" });
+    await component.clearSearch();
+    const listbox = getByRole("listbox");
+    expect(input.value).toBe("");
+    expect(listbox.querySelector(`.${highlightClass}`)).toBeNull();
+  });
+
+  test("clicking clear input button clears input value", async () => {
+    const { getByRole, getByLabelText, component } = render(Search, {
+      target,
+      props: {
+        suggestions
+      }
+    });
+    const input = getByRole("combobox") as HTMLInputElement;
+    await fireEvent.focus(input);
+    await component.$set({ searchValue: "I read the news today, oh boy." });
+    const button = getByLabelText("Clear input") as HTMLButtonElement;
+    await fireEvent.click(button);
+    expect(input.value).toBe("");
+  });
+
+  test("clicking on an option selects a suggestion", async () => {
+    const { getByRole, getAllByRole, component } = render(Search, {
+      target,
+      props: {
+        suggestions
+      }
+    });
+    const mock = jest.fn();
+    const input = getByRole("combobox") as HTMLInputElement;
+    component.$on("select", mock);
+    await component.$set({ searchValue: "" });
+    await fireEvent.focus(input);
+    await component.$set({ searchValue: "thing" });
+    const options = getAllByRole("option");
+    await fireEvent.mouseEnter(options[1]);
+    await fireEvent.click(options[1]);
+    expect(mock.mock.calls[0][0].detail).toEqual({
+      title: "Text for thing two",
+      id: "b"
+    });
+  });
+
+  test("keydown enter selects a suggestion", async () => {
+    const { getByRole, component } = render(Search, {
+      target,
+      props: {
+        suggestions
+      }
+    });
+    const mock = jest.fn();
+    const result = { title: "Text for thing one", id: "a" };
+    const input = getByRole("combobox") as HTMLInputElement;
+    component.$on("select", mock);
+    await fireEvent.focus(input);
+    await component.$set({ searchValue: "one" });
```

Comment:
I think this will always select the first result no matter what is typed if there's not an active filter applied.